apollo-router/feature_discussions.json

```diff
@@ -1,6 +1,8 @@
 {
+  "experimental": {
     "experimental_retry": "https://github.com/apollographql/router/discussions/2241",
     "experimental_response_trace_id": "https://github.com/apollographql/router/discussions/2147",
-    "experimental_logging": "https://github.com/apollographql/router/discussions/1961",
-    "experimental_jwt_authentication": "https://github.com/apollographql/router/discussions/2391"
+    "experimental_logging": "https://github.com/apollographql/router/discussions/1961"
+  },
+  "preview": {}
```

Comment:
Suggest adding a `test` gated experimental and preview dummy features for testing.
This would mean that tests dont need to change once experimental logging becomes non-experimental and would also allow unit test testing of preview code path.

Reply:
Do you mean gated on `#[cfg(test)]`?

Reply:
Yes

Reply:
I think proper tests for this feature would need to be lifecycle tests in order to capture stdout. But those use `$CARGO_BIN_EXE_router`, the "normal" executable which is compiled without `cfg(test)`.

Do we still want `experimental_noop` and `preview_noop` configs for testing, if they are user-visible?